static/js/containers/pages/DashboardPage.js

```diff
@@ -0,0 +1,97 @@
+// @flow
+/* global SETTINGS:false */
+import React from "react"
+import DocumentTitle from "react-document-title"
+import { connect } from "react-redux"
+import { createStructuredSelector } from "reselect"
+import { compose } from "redux"
+import { connectRequest } from "redux-query"
+import moment from "moment"
+
+import { DASHBOARD_PAGE_TITLE } from "../../constants"
+import {
+  enrollmentsSelector,
+  enrollmentsQuery
+} from "../../lib/queries/enrollment"
+import { formatPrettyDate, parseDateString } from "../../lib/util"
+
+import { RunEnrollment } from "../../flow/courseTypes"
+
+type DashboardPageProps = {
+  enrollments: RunEnrollment
+}
+
+export class DashboardPage extends React.Component<DashboardPageProps, void> {
```

Comment:
I think it would be better to implement this as a functional component instead using react hooks since that's what we have started to migrate to in several of our apps. A very recent example is the ocw-studio dashboard: https://github.com/mitodl/ocw-studio/blob/master/static/js/pages/SitesDashboard.tsx.

I think that would greatly simplify this since it wouldn't require `connect()` or the complexity of the `InnerDashboardPage` dance in the tests.

Reply:
@rhysyngsun That's how I set out to implement this initially, but I ran into a snag. Adding `redux-query-react` and using `useRequest` seemed to be OK, but `useSelector` was not available in this older version of `react-redux` (the version for mitxonline is `^5.0.5`, and the version for ocw-studio, just for example, is `^7.2.2`). I checked w/ Alice and I don't think there's a clean way to implement this as a functional component without upgrading `react-redux`. I can give that a try and do a smoke test of the site locally, but the jump of 2 major versions was spooky enough that I planned on just implementing this using the old `connect` style and adding a tech debt ticket to match all of these components to our more modern practices

Reply:
@rhysyngsun Fell at the first hurdle. 

![ss 2021-07-26 at 12 36 32 ](https://user-images.githubusercontent.com/14932219/127033673-0cd9fa45-d80e-4491-9a58-70a905b6af13.png)

I think this upgrade will involve a change to every component that makes a request. If it's alright w/ you, I'm going to create that tech debt ticket and leave this as-is